smac/callbacks.py

```diff
@@ -0,0 +1,16 @@
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from smac.optimizer.smbo import SMBO
+from smac.runhistory.runhistory import RunInfo, RunValue
+
+
+class IncorporateRunResultCallback:
+
```

Comment:
Could you please add some documentation about when exactly this callback is called?